hachoir/stream/input_helper.py

```diff
@@ -4,22 +4,27 @@
 
 def FileInputStream(filename, real_filename=None, **args):
     """
-    Create an input stream of a file. filename must be unicode.
+    Create an input stream of a file. filename must be unicode or a file
+    object.
 
     real_filename is an optional argument used to specify the real filename,
     its type can be 'str' or 'unicode'. Use real_filename when you are
     not able to convert filename to real unicode string (ie. you have to
     use unicode(name, 'replace') or unicode(name, 'ignore')).
     """
-    assert isinstance(filename, str)
     if not real_filename:
         real_filename = filename
     try:
-        inputio = open(real_filename, 'rb')
+        if isinstance(filename, str):
+            inputio = open(real_filename, 'rb')
+        else:
+            inputio = filename
     except IOError as err:
         errmsg = str(err)
         raise InputStreamError(
             "Unable to open file %s: %s" % (filename, errmsg))
+    filename = (filename if isinstance(filename, str)
+                else getattr(filename, 'name', ''))
```

Comment:
I would prefer to move the if inside the try block, merge it with the previous if.